Printing speedup (from an hour to half an hour)

Retropotenza opened the pull request:
* Removed the workaround for the initial vertical position (fixed in the 1.2 release of Splatoon 2)
* Added bidirectional printing (the 1.2 release of Splatoon 2 fixed also the forth to last column skip, when moving from the right to the left)
* Removed time delays on echoes (verified that there is no need for them)
* Minimized the number of echoes (verified that at least 3 needed)
* Verified and cleaned up the USB control requests and the HID task (only the latter is needed, in it now return codes for endpoints reading and writing are checked)
* Pixel skipping happens only if the Switch detect a change on its HDMI connection (this happens even if the Switch accept incoming USB reports... I don't know if is possible to avoid this issue)

LightningStalker committed:
Onboard LED(s) flash and optional buzzer sounds after printing to notify the user

LightningStalker committed:
Update README.md

Changed others to other

LightningStalker committed:
Update README.md

Fixed some of my grammar and language

LightningStalker committed:
Merge pull request #1 from LightningStalker/leds-buzzer

Onboard LED(s) flash and optional buzzer sounds after printing to notify the user

LightningStalker committed:
Added option to enable or disable LED(s) and buzzer using compiler flags

LightningStalker committed:
Added option to enable or disable LED(s) and buzzer using compiler flags

LightningStalker committed:
Resolved merge conflict by going with upstream changes

LightningStalker committed:
Fixed conflict with origin

LightningStalker committed:
Merge branch 'master' into leds-buzzer

LightningStalker committed:
Merge pull request #2 from LightningStalker/leds-buzzer

LED and buzzer function now optional

LightningStalker committed:
Update Joystick.c

Added back the missing newline at the end

LightningStalker committed:
Okay, really fixed the whitespace this time

LightningStalker committed:
Merge branch 'master' of https://github.com/LightningStalker/Splatmeme-Printer

LightningStalker committed:
Merge pull request #3 from LightningStalker/master

Fix newlines and README.md

LightningStalker committed:
Update makefile README.md with target for simple option passing & dupfix

LightningStalker committed:
Merge pull request #5 from LightningStalker/leds-buzzer

Self: looks good, self

LightningStalker committed:
Merge remote-tracking branch 'upstream/master'

LightningStalker committed:
Update Joystick.c for v1.2 since y=-1 bug was patched

Retropotenza committed:
Printing speedup

* Added bidirectional printing
* Removed time delays on echoes, there is no need for them
* Minimized the number of echoes (at least 3 needed)
* Verified and cleaned up USB control request and HID task (only the HID task is needed, return codes in it for endpoint reading and writing now are checked)

Retropotenza committed:
Printing procedures update

Retropotenza committed:
Merge pull request #1 from Retropotenza/printing-speedup

Printing speedup

undergroundmonorail commented:
This stops drawing when `ypos > 120`, but the bottom row of the image is y=119. As a result, the bottom row is overwritten with what I believe is just junk data that happens to be in ram.

I'm not super familiar with git so I don't know if I can make a pull request on your pull request or what, but changing line 224 from `if (ypos < 120)` to `if (ypos < 120 - 1)` should fix it (though I haven't tested it yet).

Retropotenza commented:
Good catch, thanks 😁. In addition to this, the code will need to ink also when in `done` state, to print the dot at `x=0, y=319`.

I'll add these fixes in this pull request (adding a slightly modified test pattern), as soon as my final test will finish.

Retropotenza committed:
Fixed an issue with final vertical position

* Moving to done when ypos reach 119
* Inking the last dot before switching to done
* Updating the test pattern to better check on bounduaries

Retropotenza committed:
Merge pull request #2 from Retropotenza/printing-speedup

Fixed an issue with final vertical position

LightningStalker commented:
Bidi, excellent!

LightningStalker commented on README.md:
```diff
@@ -12,9 +12,9 @@ Unlike the Wii U, which handles these controllers on a 'per-game' basis, the Swi
 
 #### Printing Procedure
-Just press L to select the pixel pen and plug in the controller: it will automatically sync with the console, reset the cursor position and print. In case you see issues with controller conflicts while in docked mode, try using a USB-C to USB-A adapter in handheld mode. Printing currently takes about an hour.
+Just press L to select the pixel pen and plug in the controller: it will automatically sync with the console, reset the cursor position, clean the canvas and print. In case you see issues with controller conflicts while in docked mode, try using a USB-C to USB-A adapter in handheld mode. In dock mode, changes in the HDMI connection will briefly make the Switch not responding to incoming USB comma ...
```
stop responding?

undergroundmonorail replied:
s/chanhe/change/

LightningStalker commented on Joystick.c:
```diff
@@ -197,88 +166,86 @@ void GetNextReport(USB_JoystickReport_Input_t* const ReportData) {
-			break;
-		case MOVE_DOT:
-			if (xpos < 320 - 1)
+			break;			
```
dat whitespace doe

LightningStalker committed:
Merge pull request #8 from Retropotenza/master

Bidirectional Printing

LightningStalker committed:
Fix whitespace and grammar

LightningStalker committed:
Update bin2c.py, png2c.py to add trailing newline

Retropotenza committed:
Merge pull request #3 from LightningStalker/master

Whitespace, wording in README.md

Retropotenza committed:
Added usage info to Python scripts, when called without any option

LightningStalker committed:
Merge pull request #9 from Retropotenza/master

Pulling Python script updates

shinyquagsire23 commented:
Oh nice, just let me know when you're ready for code review though since it seems there's still tweaking going on.

Retropotenza commented:
It's should be ready... there was a bug when printing the last line, fixed (other commits was minor or typos).

LightningStalker committed:
Replace missing `obj` directory

I don't know how this disappeared, but looks like it's gone from all the repos?

LightningStalker committed:
Revert "Replace missing `obj` directory"

This reverts commit 9f55e1c4caf759938fb49b33ae99c8722c3b281a.

LightningStalker commented:
Yea, it's ready for prime time.  Just printed a post yesterday morning, it's working perfectly.

The only thing I still want to add is a pause function, but that's for another day.

LightningStalker committed:
Update README.md with extra instructions for Windows

LightningStalker committed:
Added link to Atmel's website to README.md

LightningStalker committed:
Add missing code blocks to README.md

Retropotenza committed:
Merge pull request #4 from LightningStalker/master

Update README.md

LightningStalker commented:
Just a couple tweaks to README.md

shinyquagsire23 commented:
LGTM, thanks!
